app/scripts/background.js

```diff
@@ -67,8 +75,19 @@ if (inTest || process.env.METAMASK_DEBUG) {
   global.metamaskGetState = localStore.get.bind(localStore);
 }
 
-// initialization flow
-initialize().catch(log.error);
+let initialState;
+let initialLangCode;
+const initApp = (remotePort) => initialize(remotePort).catch(log.error);
+
+if (process.env.ENABLE_MV3) {
+  browser.runtime.onConnect.addListener(initApp);
```

Comment:
Can you help me understand why we would only want to do this (pass along the remotePort as an argument) in the context of MV3?  cc @jpuri

Reply:
In MV3 it is better to have event listener at root level else it is possible that first event connect is lost if user open extension immediately after installing the extension.
This parameter is required for connect method to work.

Reply:
> In MV3 it is better to have event listener at root level

Is there any reason not to have it at the root level in MV2 though? I'm wondering why we want to initialize without passing the `remotePort` in MV2?

Reply:
So I tested, removing all the feature flags for MV3 
For example:
```
let initialState;
let initialLangCode;
const initApp = (remotePort) => initialize(remotePort).catch(log.error);

// if (process.env.ENABLE_MV3) {
browser.runtime.onConnect.addListener(initApp);
(async () => {
  initialState = await loadStateFromPersistence();
  initialLangCode = await getFirstPreferredLangCode();
})();
// } else {
//   initialize().catch(log.error);
// }
```
and everything seems to work quite well in MV2. I'm wondering whether we need to make these changes for only MV3 (is my point with the comments above)

Reply:
@adonesky1 : I would prefer to keep all of these changes behind flag. Then for MV3 we can have a flask release and have these features tested more throughly before enabling for everyone.